src/java.base/share/classes/jdk/internal/util/OperatingSystem.java

```diff
@@ -129,7 +127,34 @@ public static OperatingSystem current() {
      * Names not recognized throw ExceptionInInitializerError with IllegalArgumentException.
      */
     private static OperatingSystem initOS(String osName) {
-        return OperatingSystem.valueOf(osName.toUpperCase(Locale.ROOT));
+        // Too early to use Locale conversions, manually do uppercase
+        StringBuilder sb = new StringBuilder(osName);
+        for (int i = 0; i < sb.length(); i++) {
+            char ch = sb.charAt(i);
+            if (ch >= 'a' && ch <= 'z') {
+                sb.setCharAt(i, (char)(ch - ('a' - 'A')));  // Map lower case down to uppercase
+            }
+        }
+        osName = sb.toString();
+        return OperatingSystem.valueOf(osName);
     }
 
+    /**
+     * {@return the operating system version with major, minor, micro}
+     */
+    public static Version version() {
+        return osVersion;
+    }
+
+    // Parse and save the current version
+    private static Version osVersion = initVersion();
+
+    private static Version initVersion() {
+        final String osVer = StaticProperty.osVersion();
+        try {
+            return Version.parse(osVer);
+        } catch (IllegalArgumentException ile) {
+            throw new AssertionError("os.version malformed: " + osVer);
```

Comment:
This should probably preserve the `IllegalArgumentException` stack trace:
```suggestion
        } catch (IllegalArgumentException iae) {
            throw new AssertionError("os.version malformed: " + osVer, iae);
```

And maybe throw `InternalError` instead:
```suggestion
        } catch (IllegalArgumentException iae) {
            throw new InternalError("os.version malformed: " + osVer, iae);
```